### ISSUE TITLE
Unable to copy more than the text shown on-screen

### ISSUE THREAD

**Issue by edmorley** [mozilla-frontend-infra/react-lazylog#24]
Treeherder has just switched from [unified-logviewer](https://github.com/taskcluster/unified-logviewer) to react-lazylog.

We've just had a regression bug filed:
https://bugzilla.mozilla.org/show_bug.cgi?id=1506627

> If one selects more lines than the screen can display and then tries to copy it, the clipboard will not contain all of the selected line but the last ones will be missing.
> 
> Steps to reproduce:
> 1. Go to https://treeherder.mozilla.org/logviewer.html#/jobs?job_id=211191928&repo=autoland&lineNumber=2980
> 2. Select from the TEST-START to the end of the crashing thread (lines 2973-3124).
Scroll up and down to lose the selection or hit Ctrl+C and try to paste it somewhere. The not visible part will be gone.

@eliperelman any ideas? I thought `react-lazylog` used a similar approach to `unified-logviewer` (I haven't looked at the codebases of either admitedly), so this regression was surprising.

**Comment by camd** [comment]
I'm just chiming in here since I'll own the TH bug for this going forward.  Thanks for any help you can provide.

**Comment by karlht** [comment]
Thanks, @camd.  @eliperelman, any thoughts on this at all?

**Comment by karlht** [comment]
@helfi92 Can you possibly weigh in here?

**Comment by camd** [comment]
I can reproduce this.  It seems it'll only copy the displayed lines that are selected.  I suspect it removes the react elements, so there is no true "text" that is selected beyond the bounds of the visible space.  I don't know if overscanning the loaded rows of ``react-lazylog`` would help.  But it would only mitigate it to the threshold of the overscan, if it did.

**Comment by helfi92** [comment]
Adding an `overscanCount` prop sounds like a good idea :)

**Comment by helfi92** [comment]
Wow, it turns out you can actually already do that. Internally, react-lazylog [uses the List component](https://github.com/mozilla-frontend-infra/react-lazylog/blob/49f22dc2d9f7f18b819fb231864a3a75819453e9/src/components/LazyLog/index.jsx#L12) from [`react-virtualized`](https://github.com/bvaughn/react-virtualized/blob/master/docs/List.md) and the latter seems to expose an `overscanRowCount` prop.

**Comment by helfi92** [comment]
>  I don't know if overscanning the loaded rows of `react-lazylog` would help. But it would only mitigate it to the threshold of the overscan, if it did.

Right. There's always a trade-off to make. You can try setting the prop to something like 50 or 100 and try to see if performance is still ok.

**Comment by helfi92** [comment]
It turns out `react-lazylog` sets the default `overscanRowCount` to 100. You'll want to increase that number.

**Comment by whimboo** [comment]
@camd is that something we could at least increase for now, even if it is not the final solution. Being able to c&p text which is 5 screens large would be very helpful.

**Comment by camd** [comment]
This didn't work for me.  I tried increasing the ``overscanRowCount`` to 1000.  Then when I selected line 50-60 and scrolled that just out of view, then back into view, nothing was selected.  Other times when I selected from line 100 and scrolled down with the mouse button down, it selected maybe 100 lines.  But when I scroll back up, the lines BEFORE 100 were selected.

I'm not able to get any kind of consistent line selection in ``react-lazylog``.  I think the best option is still to just go to the raw log.
